terraform/modules/active_directory_dns_resolver/main.tf

```diff
@@ -0,0 +1,33 @@
+#
+# Configure DHCP for the VPC to return the Active Directory DNS servers.
+# Those servers are then in turn configured to forward requests to the Amazon provided VPC DNS (See active_directory module README)
+#
+# This means the AD domain (dluhcdata.local) will resolve correctly through AD.
+#
+
+variable "ad_dns_server_ips" {
+  type = list(string)
+}
+
+variable "vpc" {
+  type = object({
+    id         = string
+    cidr_block = string
+  })
+}
+
+locals {
+  # Always at VPC CIDR base + 2, e.g. 10.0.0.2
+  amazon_provided_vpc_dns = cidrhost(var.vpc.cidr_block, 2)
+  # Add the Amazon provided DNS server to the end of the list as a backup in case AD doesn't respond
+  dns_servers = concat(var.ad_dns_server_ips, [local.amazon_provided_vpc_dns])
```

Comment:
You can use the string "AmazonProvidedDNS" instead of calculating vpc cidr + 2.

I did that originally but then saw a recommendation against it on this page https://docs.aws.amazon.com/vpc/latest/userguide/DHCPOptionSet.html

```You can enter either AmazonProvidedDNS or custom domain name servers. Using both might cause unexpected behavior```

I don't know what the issue would be though. Maybe the order it tries them isn't guaranteed so it's theoretically possible they'd resolve a domain differently? I agree having a fallback seems safer, so maybe it outweighs the vague risk of "unexpected behaviour".

Reply:
I'm not sure what it means either. I'll switch to using "AmazonProvidedDNS" (going to make a markups PR)